datalake/src/main/java/com/sequenceiq/datalake/flow/datalake/scale/DatalakeHorizontalScaleActions.java

```diff
@@ -0,0 +1,223 @@
+            }
+
+            @Override
+            protected Object getFailurePayload(DatalakeHorizontalScaleSdxEvent payload, Optional<SdxContext> flowContext, Exception ex) {
+                var sdxCluster = sdxService.getById(payload.getResourceId());
+                String message = ex.getMessage();
+                eventSenderService.sendEventAndNotification(sdxCluster, ResourceEvent.DATALAKE_HORIZONTAL_SCALE_VALIDATION_FAILED,
+                        List.of(message));
+                return new DatalakeHorizontalScaleSdxEvent(DATALAKE_HORIZONTAL_SCALE_FAILED_EVENT.selector(),
+                        payload.getResourceId(),
+                        payload.getSdxName(),
+                        payload.getUserId(),
+                        payload.getResourceCrn(),
+                        payload.getScaleRequest(),
+                        ex,
+                        payload.getFlowId(),
+                        payload.getCommandId());
+            }
+        };
+    }
+
+    @Bean(name = "DATALAKE_HORIZONTAL_SCALE_START_STATE")
+    public Action<?, ?> datalakeHorizontalScaleStart() {
+        return new AbstractSdxAction<>(DatalakeHorizontalScaleSdxEvent.class) {
+            @Override
+            protected void doExecute(SdxContext context, DatalakeHorizontalScaleSdxEvent payload, Map<Object, Object> variables) {
+                var sdxCluster = sdxService.getById(payload.getResourceId());
+                sdxStatusService.setStatusForDatalake(DatalakeStatusEnum.DATALAKE_HORIZONTAL_SCALE_IN_PROGRESS,
+                        "Horizontal Scale In Progress", sdxCluster);
+                eventSenderService.sendEventAndNotification(sdxCluster, DATALAKE_HORIZONTAL_SCALE_IN_PROGRESS);
+                StackScaleV4Request scaleRequest = new StackScaleV4Request();
+                scaleRequest.setGroup(payload.getScaleRequest().getGroup());
+                scaleRequest.setDesiredCount(payload.getScaleRequest().getDesiredCount());
+                scaleRequest.setForced(false);
+                scaleRequest.setStackId(sdxCluster.getStackId());
+                scaleRequest.setAdjustmentType(AdjustmentType.EXACT);
+                LOGGER.info("Horizontal scale starts in group of {} with desired count {}", scaleRequest.getGroup(), scaleRequest.getDesiredCount());
+                String flowId = null;
+                try {
+                    flowId = ThreadBasedUserCrnProvider.doAsInternalActor(
+                            regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+                            () -> stackV4Endpoint.putScaling(WORKSPACE_ID, sdxCluster.getClusterName(), scaleRequest, sdxCluster.getAccountId())
+                    ).getPollableId();
+                } catch (NotFoundException e) {
+                    LOGGER.info("Cannot find stack on Cloudbreak side {}", sdxCluster.getClusterName());
```

Comment:
is this a desired behavior - if stack not found then continue flow? No exception thrown - looks suspicious